app/controllers/products_controller.rb

```diff
@@ -2,28 +2,48 @@ class ProductsController < ApplicationController
   before_action :set_product, only:[:show, :edit, :update, :destroy]
   
   def index
+    @products = Product.all
   end
-  
+
   def show
   end
-  
+
   def dashboard
+    @products = Product.all
   end
-  
+
   def new
+    @product = Product.new
   end
 
   def create
+    product = Product.create!(product_params.except(:image))
+    product.image.attach(product_params[:image])
+    redirect_to dashboard_path
```

Comment:
add the empty line above redirect. It is more readable that way. in the first 2 lines, you created the product and you attached the image and on the next line, you redirect the user.